src/test/java/io/xstefank/wildfly/bot/utils/GitHubJson.java

```diff
@@ -0,0 +1,82 @@
+package io.xstefank.wildfly.bot.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Class responsible for handling and changing properties of the GitHub JSON template file.
+ */
+public class GitHubJson {
+
+    private static String PULL_REQUEST = "pull_request";
+    private static String COMMITS = "commits";
+    private static String BODY = "body";
+    private static String TITLE = "title";
+    private static String SHA = "sha";
+    private static String HEAD = "head";
+    private static String ACTION = "action";
+
+    private static JsonNode file;
+    private static Builder builder;
+
+    private GitHubJson(Builder jsonHandlerBuilder) {
+        builder = jsonHandlerBuilder;
+        file = jsonHandlerBuilder.jsonFile;
+    }
+
+    public static Builder builder(String fileName) throws IOException {
+        return new Builder(fileName);
+    }
+
+    public String commitSHA() {
+        return file.get(PULL_REQUEST).get(HEAD).get(SHA).textValue();
+    }
+
+    public String jsonString() {
+        return file.toPrettyString();
+    }
+
+    public static final class Builder {
+
+        private final JsonNode jsonFile;
+        private final ObjectMapper objectMapper = new ObjectMapper();
+
+        private Builder(String fileName) throws IOException {
+            File jsonFile = new File("src/test/resources/" + fileName);
+            if (!jsonFile.exists()) {
+                throw new FileNotFoundException("File " + fileName + " cannot be found or does not exist.");
+            }
+            this.jsonFile = objectMapper.readTree(jsonFile);
+        }
+
+        public Builder action(Action action) {
+            ((ObjectNode) this.jsonFile).put(ACTION, action.getValue());
+            return this;
+        }
+
+        public Builder pullRequestTitle(String title) {
+            ((ObjectNode) this.jsonFile.get(PULL_REQUEST)).put(TITLE, title);
+            return this;
+        }
+
+        public Builder pullRequestBody(String body) {
+            ((ObjectNode) this.jsonFile.get(PULL_REQUEST)).put(BODY, body);
+            return this;
+        }
+
+        public Builder commitQuantity(int quantity) {
```

Comment:
not for now, but maybe later we can replace this with .commit(String).commit(String) as we discussed